core/matrix/csr.cpp

```diff
@@ -136,6 +137,35 @@ void Csr<ValueType, IndexType>::move_to(Dense<ValueType> *result)
 }
 
 
+template <typename ValueType, typename IndexType>
+void Csr<ValueType, IndexType>::convert_to(
+    Sellp<ValueType, IndexType> *result) const
+{
+    auto exec = this->get_executor();
+    const auto stride_factor = (result->get_stride_factor() == 0)
+                                   ? default_stride_factor
+                                   : result->get_stride_factor();
+    const auto slice_size = (result->get_slice_size() == 0)
+                                ? default_slice_size
+                                : result->get_slice_size();
+    size_type total_columns = 0;
+    exec->run(csr::make_calculate_total_cols(this, &total_columns,
+                                             stride_factor, slice_size));
+    const auto total_cols = std::max(result->get_total_cols(), total_columns);
```

Comment:
These are all needed to calculate how much memory has to be allocated for a sellp matrix. The only one of the three which actually is influenced by the number of entries of the matrix is `total_cols`. It is the sum over the lengths of the rows in each slice, so we store `total_cols` * `slice_size` elements.

Reply:
I still don't quite get the `std::max` for `total_cols` here. Best case is `result->get_total_cols()` is identical to (or lower than) `total_columns`, so we do not waste memory. If it is larger, we allocate more memory on the GPU than necessary (since we don't reuse the memory from `result`).

Therefore, I think it is best to remove this `std::max` and just use the result from `csr::make_calculate_total_cols`.
The others make complete sense because it allows the user to change these parameters.